fix: Avoid internal error when logging in with the wrong account to verify email address

come-nc opened the pull request:
* Resolves: # <!-- related github issue -->

## Summary

Avoid internal error when logging in with the wrong account to verify email address

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

come-nc committed:
fix: Avoid internal error when logging in with the wrong account to verify email address

Signed-off-by: Côme Chilliet <come.chilliet@nextcloud.com>

susnux commented on apps/provisioning_api/lib/Controller/VerificationController.php:
```diff
@@ -51,11 +51,18 @@ public function __construct(
+		} catch (\Exception $e) {
+			return new TemplateResponse(
+				'core', 'error', [
+					'errors' => [['error' => $e->getMessage()]]
```
basically two reasons:
- Do not show errors that potentially leaking internals to random users
- Have proper translated errors for users but keep English in the logs for debugging :rescue_worker_helmet: tickets

come-nc committed:
fix: Use HintException instead of InvalidArgumentException

To carry translated error messages intended for the end user,
 HintException is the correct class.

Signed-off-by: Côme Chilliet <come.chilliet@nextcloud.com>

come-nc commented:
/backport to stable32
